### PR TITLE
Add initial tests for the model

### PR DESCRIPTION
**Pull request by djsauble**
This PR adds unit tests for all parts of the model except those relating to pyspark.

It also refactors the scraper tests so they can all be run from the `pipeline/` folder with a simple `python3 -m unittest`.

No functionality has changed, just tests. It looks a little scarier than it is.

### CODE REVIEW

File: pipeline/client.py [hunk 1]
```diff
@@ -19,47 +19,49 @@
...
+  package_a INTEGER REFERENCES packages (id),
+  package_b INTEGER REFERENCES packages (id),
+  similarity FLOAT(4) NOT NULL,
+  bounded_similarity INTEGER,
+  CONSTRAINT unique_pkg_to_pkg UNIQUE (package_a, package_b)
+);
+CREATE INDEX ON similarity (package_a);
+CREATE INDEX ON similarity (package_b);
+CREATE INDEX ON similarity (similarity);
     """
```
Comment (luke777gu) [inline pipeline/client.py:63]: I did not pay attention when Roman mentioned on this query commands. He probably said that we should not directly use a very long string like this, right?

File: pipeline/tests/test_model.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            {
+                'id': 1,
+                'name': '\'pkg:npm/countup.js@2\'',
+                'monthly_downloads_last_month': 400451,
+                'monthly_downloads_a_year_ago': 0,
+                'categories': 'null',
+                'modified': '\'2019-03-14 12:42:34.846-07\'',
+                'retrieved': '\'2020-04-25 19:03:37.409069-07\'',
+                'app_ids': app_data[-1:]
+            },
```
Comment (luke777gu) [inline pipeline/tests/test_model.py:34]: We probably need to create more models for pypi and condo in the end. It should be painful. LOL

File: pipeline/model/model.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    Compute a similarity matrix from the dependency table
+    :param cursor: Database cursor
+    :return: Spark DataFrame
+    """
+
+    SC = SparkContext("local[1]", "pkgpkr")
+
+    # Load the raw data into Spark
+    cursor.execute("SELECT * FROM dependencies")
+    DEPENDENCIES = cursor.fetchall()
```
Comment (vpopil) [inline pipeline/model/model.py:24]: why all of those are capitalized (thru line63? none are constants, is it something pyspark specific style?

File: pipeline/run_scraper.py [hunk 4]
```diff
@@ -22,16 +22,16 @@ def main():
 
     # Fetch applications from GitHub for javascript
     for i in range(0, 30 * int(os.environ['MONTH'])):
-        print("Fetching day " + str(i) + " for JS")
-        run_query(today - datetime.timedelta(days=i))
+         print("Fetching day " + str(i) + " for JS")
+         run_query(today - datetime.timedelta(days=i))
```
Comment (vpopil) [inline pipeline/run_scraper.py:26]: seems over-indented (i.e 5)?  also maybe string interpolation for consistency?

File: pipeline/train_model.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    # Assert that the necessary environment variables are present
+    assert USER, "DB_USER not set"
+    assert PASSWORD, "DB_PASSWORD not set"
+    assert HOST, "DB_HOST not set"
+    assert DATABASE, "DB_DATABASE not set"
+    assert PORT, "DB_PORT not set"
+
+    # Connect to the database
+    DB = psycopg2.connect(CONN_STRING)
+    CUR = DB.cursor()
```
Comment (vpopil) [inline pipeline/train_model.py:29]: same here, why all upper case

File: pipeline/client.py (hunk 1 above)
Comment (djsauble) [reply pipeline/client.py]: I've refactored this into a separate `.sql` provisioning file.

File: pipeline/model/model.py (hunk 3 above)
Comment (djsauble) [reply pipeline/model/model.py]: Fixed

File: pipeline/train_model.py (hunk 5 above)
Comment (djsauble) [reply pipeline/train_model.py]: Fixed

File: pipeline/tests/test_model.py (hunk 2 above)
Comment (djsauble) [reply pipeline/tests/test_model.py:34]: Agreed. More tests would be good.

**Pull request merged**